docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6328,409 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+  properties under section [GenericTable categories]
+  (#sec-generic-table-categories) for more details.
+
+* `repeated GenericTableInstance`, contains the info for one instance of
+  a generic-table-type. All tables of one type are grouped under this.
+
+GenericTableInstance
+* `preamble`, a `Preamble` message with the ID, name, and alias of this
+  `GenericTable` instance.
+
+* `generic_match_fields`, a repeated field of type `GenericMatchField`
+  representing the data to be used to construct the lookup key matched in this
+  table. For information check the generic_match_fields info in the [Generic
+  Table section](#sec-generic-table-entry).
+
+* `union_refs`, a repeated `UnionRef` field representing the set of possible
+  union members for this table. Functionally, it behaves same as that of
+  `ActionRefs`.
+  Please check the union_refs info in the [Generic Table section]
+  (#sec-generic-table-entry).
+
+* `const_default_union_id`, if this table has a constant default union, this
+  field will carry the `uint32` identifier of that action, otherwise its value
+  will be 0.It is similar to a default action for Match Tables. Being constant
+  means that the control plane cannot set a different default union at runtime
+  or change the default union's fields.
+
+* `size`, an `int64` describing the desired number of table entries that the
+  target should support for the table.  See the "Size" subsection within the
+  "Table Properties" section of the P4~16~ language specification for details
+  [@P4TableProperties].
+
+* `other_properties`, an `Any` Protobuf message [@ProtoAny] to embed
+  architecture-specific or target-specific table properties [@P4TableProperties]
+  that the target wants to convey.
+
+## `GenericData` { #sec-p4data-generic-data}
+
+`GenericData` is an addition to p4data.proto- which adds support for
+data types which are not covered by P4Data. This incorporates all the data types
+used as primitive data types like bytes, strings.
+Some of them have default values. These default values are only applicable
+when in a Union field and not for Match fields. Match fields do not have
+default values because if a match field is not present, then it is considered
+to be a wildcard match. But if a union member is not present, then the default
+value is used. Numeric Default values have been kept as `bytes` so that
+lengths > 64 can be easily accommodated.
+
+  * bitstring : Bytes type used for simple unsigned integers or bytes. Default
+    value applicable. If not present, it is 0.
+  * bool : Boolean type values. Default value applicable. If not present, it
+    is `false`.
+  * generic_struct : Used to define a structure that can contain
+    one or more of the other members as named fields. One structure can contain
+    different members of different types in GenericData.
+  * generic_list : Used to define a list of values with the same type. It can
+    also be a list of generic_structs or generic_list.
```

Comment:
Can it be a list of elements, each of which is a generic_unordered_set?  If so, why only mention generic_struct or generic_list here?  If the element type can be any GenericData type, it seems best to simply say that.